src/components/views/rooms/NotificationBadge/StatelessNotificationBadge.tsx

```diff
@@ -70,17 +70,27 @@ export const StatelessNotificationBadge = forwardRef<HTMLDivElement, XOR<Props,
             symbol = formatCount(count);
         }
 
+        // We show a dot if either:
+        // * The props force us to, or
+        // * It's just an activity-level notification or (in theory) lower and the room isn't knocked
+        const badgeType =
+            forceDot || (level <= NotificationLevel.Activity && !knocked)
```

Comment:
so previously we would show a `dot` if count was 0 (I think?) but now we will show a badge in this case? I think?

Is that deliberate?

Reply:
Yes: because there is no unread message that triggers a notification, the count will be zero for rooms marked as unread, but we still want to display a badge.